crypto/evp/evp_enc.c

```diff
@@ -354,8 +354,10 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
 
         case EVP_CIPH_CBC_MODE:
             n = EVP_CIPHER_CTX_get_iv_length(ctx);
-            if (!ossl_assert(n >= 0 && n <= (int)sizeof(ctx->iv)))
-                    return 0;
+            if (n <= 0 || n > (int)sizeof(ctx->iv)) {
```

Comment:
The code allowed 0 length iv before. Not sure if we should change it.

Reply:
Doesn't a zero length IV lead to an undefined memcpy(3) a few lines later?

Reply:
no only if iv == NULL at the same time.

Reply:
Removed the equality test.